packages/clarity-js/src/data/extract.ts

```diff
@@ -0,0 +1,139 @@
+import { ExtractSource, Syntax, Type } from "@clarity-types/core";
+import { Event, Setting } from "@clarity-types/data";
+import config from "@src/core/config";
+import encode from "./encode";
+import * as dom from "@src/layout/dom";
+import * as internal from "@src/diagnostic/internal";
+import { Code, Constant, Severity } from "@clarity-types/data";
+
+export let data: {[key: number]: string } = {};
+export let updateKeys: (number | string)[] = [];
+
+let variables : { [key: number]: Syntax[] } = {};
+let selectors : { [key: number]: string } = {};
+
+export function start(): void {
+    let e = config.extract;
+    if (!e) { return; }
+    for (let i = 0; i < e.length; i+=3) {
```

Comment:
>for (let i = 0; i < e.length; i+=3) { [](http://example.com/codeflow?start=4&length=37)

Since this depends on configuration - which could technically come in corrupted format - should we wrap it in try / catch like before to ensure we do not break the script for wrong inputs?